x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/original_scenarios.md

```diff
@@ -0,0 +1,264 @@
+When users navigate to the Rules Management Page
+Then users should see an update callout message
+And the number of outdated rules should be displayed on the Rules Updates tab
+```
+
+#### **Scenario: User is notified of available prebuilt rules to install when a rule is deleted**
+
+```Gherkin
+Given user has the latest version of prebuilt rules installed
+When user navigates to Rules Management Page
+And user deletes some prebuilt rules
+Then user should see a callout message with the same amount of prebuilt rules ready to install
+And the updated number of available rules to install should be displayed on the +Add Elastic Rules link
+```
+
+#### **Scenario: User is notified that all available rules have been installed in the Add Elastic Rules**
+
+```Gherkin
+Given user has all available rules installed
+When user navigates to the Add Elastic Rules Page
+Then user should see a message indicating that all available rules have been installed
+And user should see a CTA that leads to the Rules Management Page
+And user should navigate back to Rules Management Page when clicking on the CTA
+```
+
+#### **Scenario: User is notified that all installed rules are up to date in the Rule Updates tab**
+
+```Gherkin
+Given user has all available rules installed
+When user navigates to the Rule Update
+Then user should see a message indicating that all installed rules are up to date
+```
+
+### Prebuilt Rules Installation
+
+#### **Scenario: User without any installed prebuilt rules can install `<amount>` prebuilt rules**
+
+```Gherkin
+Given a user that doesn’t have prebuilt rules installed
+When user navigates to Add Elastic Rules Page
+Then available prebuilt rules are displayed on Elastic Rules table
+And user can install <amount> prebuilt Rules
+And successfully installed message is displayed after installation
+And installed rules are removed from Elastic Rules table
+And rules to install counter is decreased accordingly
+
+Examples:
+  | amount     |
+  | all        |
+  | selected   |
+  | individual |
+```
+
+#### **Scenario: User navigating to the Add Elastic Rules page sees a loading skeleton until the prebuilt rules package installation is completed**
+
+```Gherkin
+Given a user that is on Rules Management Page
+When user to the Add Elastic Rules page before rules package is installed
+Then a loading skeleton is displayed until the installation is completed
+```
+
+### Prebuilt Rules Update
+
+#### **Scenario: Users can update prebuilt rules**
+
+```Gherkin
+Given user already has 8.x prebuilt rules installed in Kibana
+And there are new updates available for those prebuilt rules
+And user is on Rules Management Page
+When user navigates to the Rules Update tab
+Then user should see all the prebuilt rules that have updates available
+And user can update outdated prebuilt rules
+And successfully updated message is displayed
+And Rules Upgrade tab counter is decreased according to the number of updated rules
+```
```

Comment:
Split into 3 scenarios to be able to add more details into them:

#### **Scenario: User can upgrade prebuilt rules one by one**

**Coverage**: 1 e2e test with mock rules.

```Gherkin
Given X prebuilt rules are installed in Kibana
And for Y of the installed rules there are new versions available
And user is on the Rule Management page
When user opens the Rule Updates table
Then Y rules available for upgrade should be displayed in the table
When user upgrades one individual rule
Then success message should be displayed after upgrade
And the upgraded rule should be removed from the table
And user should see the number of rules available to upgrade decreased by 1
```

#### **Scenario: User can upgrade multiple prebuilt rules selected on the page**

**Coverage**: 1 e2e test with mock rules.

```Gherkin
Given X prebuilt rules are installed in Kibana
And for Y of the installed rules there are new versions available
And user is on the Rule Management page
When user opens the Rule Updates table
Then Y rules available for upgrade should be displayed in the table
When user selects <Z> rules
Then user should see a CTA to upgrade <Z> number of rules
When user clicks the CTA
Then success message should be displayed after upgrade
And all the <Z> upgraded rules should be removed from the table
And user should see the number of rules available to upgrade decreased by <Z> number of upgraded rules

Examples:
  | Z                               |
  | a few rules on the page, e.g. 2 |
  | all rules on the page, e.g. 12  |
```

#### **Scenario: User can upgrade all available prebuilt rules at once**

**Coverage**: 1 e2e test with mock rules.

```Gherkin
Given X prebuilt rules are installed in Kibana
And for Y of the installed rules there are new versions available
And user is on the Rule Management page
When user opens the Rule Updates table
Then Y rules available for upgrade should be displayed in the table
When user upgrades all rules
Then success message should be displayed after upgrade
And user should NOT see a CTA to upgrade prebuilt rules
And user should NOT see a number of rules available to upgrade
And user should NOT see the Rule Updates table
```